xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/suggest/suggestPages.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+require.config({
+  paths: {
+    'xwiki-selectize': "$xwiki.getSkinFile('uicomponents/suggest/xwiki.selectize.js', true)" +
+      "?v=$escapetool.url($xwiki.version)"
+  }
+});
+
+define('xwiki-suggestPages', ['jquery', 'xwiki-selectize'], function($) {
+  'use strict';
+
+  var pageIcon = $jsontool.serialize($services.icon.getMetaData('page_white'));
+
+  var getSelectizeOptions = function(select) {
+    return {
+      create: true,
+      load: function(text, callback) {
+        loadPages(text).done(function(data) {
+          var pages = [];
+          data.searchResults.forEach(function (element) {
+            var label = element.title;
+            var queryString = "";
+            // Separate spaces with " / " and unescape characters. E.g. A\.B.C\\D => A.B / C\D
+            var hint = element.space.replace(/([^\\])\./g, "$1 / ").replace(/\\(.)/g, "$1");
+            if (element.language) {
+              queryString = "language=" + element.language;
+              hint += " (" + element.language + ")";
+            }
+            var url = new XWiki.Document(XWiki.Model.resolve(element.id, XWiki.EntityType.DOCUMENT))
+              .getURL("view", queryString);
+            pages.push({
+              'value': element.pageFullName,
+              'label': label,
+              'icon': pageIcon,
+              'url': url,
+              'hint': hint
+            });
+          });
+          callback(pages);
+        }).fail(callback);
+      }
+    }
+  };
+
+  var loadPages = function(text) {
+    // We need to escape backslashes for Solr.
+    text = text.replace(/\\/g, "\\\\");
+    var response = $.getJSON("${request.contextPath}/rest/wikis/query/", {
```

Comment:
I guess the REST resource that uses database search is not good enough?

Reply:
@mflorea I thought we wanted to use the Solr search, didn't we? You want me to try using [/wikis/{wikiName}/search](https://www.xwiki.org/xwiki/bin/view/Documentation/UserGuide/Features/XWikiRESTfulAPI#H2Fwikis2F7BwikiName7D2Fsearch3Fq3D7Bkeywords7D5B5B26scope3D7Bname2Ccontent2Ctitle2Cobjects7D...5D26start3Dn5D5B26number3Dn5D5B26orderField3Dfield26order3D7Basc2Cdesc7D5D5Bdistinct3D7Btrue2Cfalse7D5D5B26prettyNames3D7Btrue2Cfalse7D5D)?

Reply:
In a previous comment I suggested you to try to use the same REST resource as the "Go To Page" (Ctrl + G) feature. Which is:

    ${request.contextPath}/rest/wikis/__wiki__/search?scope=name&number=10

I don't know if it's possible or easy, but I would investigate it at least.